### PR TITLE
Add spec for removing `type/` from Record ID

### PR DESCRIPTION
**Pull request by cheungpat**
connects #126

### CODE REVIEW

File: features/126-record-id/api.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Record Type - "note"
+Record ID - "0ebd484c-ed82-4b04-aba7-33fc530f3c88"
+
+### Changes to SDK API
+
+* API that accepts deprecated ID as parameter should accept a Record Type and
+  a Record ID as parameter, both of `string` type.
+* API that accepts a list of deprecated IDs as parameter should accept a Record
+  Type (string) and a list of Record ID (string).
+* API that returns deprecated IDs (or list) should return Record ID (string),
```
Comment (chpapa) [inline features/126-record-id/api.md]: similarly, have we got an overview of our api, that when shall we return object or ID?

File: features/126-record-id/api.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  the Record Type is either
+  * Will be omitted if Record Type is already known. For example, delete
+    operation will return a Record IDs, and the Record Type is inferred from
+    the Record Type used in the delete operation.
+  * Will return a Reference instead, which has separate Record ID and Record
+    Type fields.
+  * Other cases will need to be discussed.
+* API that supports deprecated IDs with multiple Record Types will no longer
```
Comment (chpapa) [inline features/126-record-id/api.md]: just to confirm the only APIs fall into this category are `fetch` and `delete`

File: features/126-record-id/api.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	"_transient": {"secret":null}
+}
+```
+
+New:
+
+```json
+{
+	"_recordID": "b7c66af9-cabd-4e12-bcce-0a873dfb6328",
+	"_recordType": "note",
```
Comment (chpapa) [inline features/126-record-id/api.md]: In last meeting we discussed, we might want to have `_id` for compatibility, and a reference data point for the decision is how many Oursky developers are using it, have we asked or researched a bit (Say, do a quick search of relevant API across oursky GitHub repos)

File: features/126-record-id/implementation.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+```javascript
+var Note = Record.extend('note');
+var note = new Note({});
+
+// Old
+console.log(note.id);  // 'note/91c02d33-ea8a-41d3-ae6c-a87e064eaa5c'
+console.log(note.recordType);  // 'note'
+
+// New
+console.log(note.recordID);  // '91c02d33-ea8a-41d3-ae6c-a87e064eaa5c'
```
Comment (chpapa) [inline features/126-record-id/implementation.md]: Is it intended to rename `id` to `recordID`? Or it is an unconscious choice?

File: features/126-record-id/implementation.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+The Record fetch and delete APIs are changed substantially:
+
+```obj-c
+// Old
+- (void)fetchRecordWithID:(SKYRecordID *)recordID
+        completionHandler:(void (^)(SKYRecord *record, NSError *error))completionHandler;
+
+// New
+- (void)fetchRecordWithID:(NSString *)recordID
```
Comment (chpapa) [inline features/126-record-id/implementation.md]: Is there any reason to change SKYRecordID to string? (Just asking, no thought from me)

File: features/126-record-id/api.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+Throughout this document the different ID formats are described as:
+
+Deprecated ID - "note/0ebd484c-ed82-4b04-aba7-33fc530f3c88"
+Record Type - "note"
+Record ID - "0ebd484c-ed82-4b04-aba7-33fc530f3c88"
+
+### Changes to SDK API
+
+* API that accepts deprecated ID as parameter should accept a Record Type and
```
[the remark this answers is not in the document]
Comment (cheungpat) [reply features/126-record-id/api.md]: I think we will handle this with the cross platform API consistency issue. I think I am with you that for parameters that take ID, we should also take object.

File: features/126-record-id/api.md (hunk 1 above)
Comment (cheungpat) [reply features/126-record-id/api.md]: That depends on the API?

fetchRecord: ID > Record
saveRecord: Record > Record
deleteRecord: (ID or Record) > ID (returning an object after deleting it doesn’t seem intuitive)

File: features/126-record-id/api.md (hunk 2 above)
Comment (cheungpat) [reply features/126-record-id/api.md]: Yes.

File: features/126-record-id/api.md (hunk 3 above)
Comment (cheungpat) [reply features/126-record-id/api.md]: I didn’t. Let me try ask around.

On the protocol level, I think we will have `_id` for compatibility. For JS SDK, I think keeping `_id` will confuse people, so we should remove that.

File: features/126-record-id/implementation.md (hunk 5 above)
Comment (cheungpat) [reply features/126-record-id/implementation.md]: My idea is that using string ID is easy to understand for most people. For new learner, they don’t need to figure out how to construct a `SKYRecordID`.

I am open to suggestion for change here, as having `SKYRecordID` might help type safety.

File: features/126-record-id/implementation.md (hunk 4 above)
Comment (cheungpat) [reply features/126-record-id/implementation.md]: It is renamed to recordID to keep consistency with other platforms. Android SDK has `recordID` as well.

**Comment by cheungpat** [comment]
I have updated the documents will minor changes.

When the SDK has made changes to separate record type and record ID information, I think it is beneficial if we have the same separation on the server-side API as this will make the two APIs easier to understand. I understand that it is not a requirement for the server-side API to match the client-side API, but I don’t see why matching both sets of API is a bad thing. That’s why I will keep the record type and record ID separated in the payload.

An effort will be made to keep both formats in the Record Payload format, so that old clients are still compatible with new version of server. In the long run, we will remove the deprecated fields.

I think this is the best I can come up with, having considered the opinions of the relevant parties. If you have a better idea. Please feel free to make a pull request and I will be happy to include them.

**Comment by chpapa** [comment]
I don't have any more comment at this stage, thanks!

**Pull request merged**